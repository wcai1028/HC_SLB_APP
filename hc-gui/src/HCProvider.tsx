import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
