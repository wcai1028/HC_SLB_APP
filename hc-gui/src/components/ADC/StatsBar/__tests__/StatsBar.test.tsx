import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import * as moxios from 'moxios'
import HCProvider from 'src/HCProvider'

const renderTestComponent = () => {
  const StatusBar = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <StatusBar />
    </HCProvider>,
  )
}

beforeEach(() => {
    moxios.install()
})
afterEach(() => {
    moxios.uninstall()
})

describe('should render properly', () => {
    test('with EMPTY Attribute', (done: jest.DoneCallback) => {
      const testComponent = renderTestComponent()
      // wait all component are rendered ready
      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })
})