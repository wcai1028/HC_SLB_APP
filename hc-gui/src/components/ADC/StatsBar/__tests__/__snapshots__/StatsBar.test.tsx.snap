// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with EMPTY Attribute 1`] = `
<div
  className="ant-row-flex ant-row-flex-center ant-row-flex-middle statsBar"
>
  <div
    className="ant-col-4"
  >
    <div>
      Average
    </div>
    <div>
      Health
    </div>
  </div>
  <div
    className="ant-col-4"
  >
    <div>
      3
      <span
        className="statsUnit"
      >
        Gpbs
      </span>
    </div>
    <div>
      Throughput
    </div>
  </div>
  <div
    className="ant-col-4"
  >
    <div>
      <div>
        20
        <span
          className="statsUnit"
        >
          K
        </span>
      </div>
      <div>
        Connections
      </div>
    </div>
  </div>
  <div
    className="ant-col-4"
  >
    <div>
      <div>
        121
        <span
          className="statsUnit"
        >
          K
        </span>
      </div>
      <div>
        Current Connections
      </div>
    </div>
  </div>
  <div
    className="ant-col-4"
  >
    <div>
      <div>
        1
        <span
          className="statsUnit"
        >
          K/S
        </span>
      </div>
      <div>
        Connection Rate
      </div>
    </div>
  </div>
  <div
    className="ant-col-4"
  >
    <div>
      <div>
        96
      </div>
      <div>
        Errors
      </div>
    </div>
  </div>
</div>
`;
